contracts/ModifiedTokenDistro.sol

```diff
@@ -0,0 +1,339 @@
+        require(!hasRole(DISTRIBUTOR_ROLE, recipient), 'TokenDistro::claimableAt: DISTRIBUTOR_ROLE_CANNOT_CLAIM');
+        require(timestamp >= getTimestamp(), 'TokenDistro::claimableAt: NOT_VALID_PAST_TIMESTAMP');
+        uint256 unlockedAmount = (globallyClaimableAt(timestamp) * balances[recipient].allocatedTokens) / totalTokens;
+
+        return unlockedAmount - balances[recipient].claimed;
+    }
+
+    /**
+     * Function to get the unlocked tokens for a specific address. It uses the current timestamp
+     * @param recipient account to query
+     */
+    function claimableNow(address recipient) public view override returns (uint256) {
+        return claimableAt(recipient, getTimestamp());
+    }
+
+    /**
+     * Function that allows the DEFAULT_ADMIN_ROLE to change a recipient in case it wants to cancel an allocation
+     * @dev The change can only be made when cancelable is true and to an address that has not previously received
+     * an allocation and the distributor cannot change its address
+     *
+     * Emits a {ChangeAddress} event.
+     *
+     */
+    function transferAllocation(address prevRecipient, address newRecipient) external override {
+        require(cancelable, 'TokenDistro::transferAllocation: NOT_CANCELABLE');
+
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), 'TokenDistro::transferAllocation: ONLY_ADMIN_ROLE');
+        _transferAllocation(prevRecipient, newRecipient);
+    }
+
+    /**
+     * Function to claim tokens for a specific address. It uses the current timestamp
+     *
+     * Emits a {claim} event.
+     *
+     */
+    function _claim(address recipient) private {
+        uint256 remainingToClaim = claimableNow(recipient);
+
+        require(remainingToClaim > 0, 'TokenDistro::claim: NOT_ENOUGH_TOKENS_TO_CLAIM');
+
+        balances[recipient].claimed = balances[recipient].claimed + remainingToClaim;
+
+        token.safeTransfer(recipient, remainingToClaim);
+
+        emit Claim(recipient, remainingToClaim);
+    }
+
+    /**
+     * Function to change the duration
+     */
+    function setDuration(uint256 newDuration) public {
+        require(hasRole(DEFAULT_ADMIN_ROLE, msg.sender), 'TokenDistro::setDuration: ONLY_ADMIN_ROLE');
+
+        require(startTime > getTimestamp(), 'TokenDistro::setDuration: IF_HAS_NOT_STARTED_YET');
+
+        duration = newDuration;
+
+        emit DurationChanged(newDuration);
+    }
+
+    function _transferAllocation(address prevRecipient, address newRecipient) internal {
+        require(
+            balances[prevRecipient].allocatedTokens > 0, 'TokenDistro::transferAllocation: NO_ALLOCATION_TO_TRANSFER'
+        );
+        require(
+            !hasRole(DISTRIBUTOR_ROLE, prevRecipient) && !hasRole(DISTRIBUTOR_ROLE, newRecipient),
+            'TokenDistro::transferAllocation: DISTRIBUTOR_ROLE_NOT_A_VALID_ADDRESS'
+        );
+        // balance adds instead of overwrites
+        balances[newRecipient].allocatedTokens =
```

Comment:
is there a possibility that balances[prevRecipient].allocatedTokens == balances[prevRecipient].claimed ? If yes lets check it before.